hopsworks-IT/src/test/ruby/spec/featurestore_spec.rb

```diff
@@ -166,10 +167,104 @@
+
+        it "should be able to add s3 connector to the featurestore with encryption algorithm and with encryption key" do
+          project = get_project
+          encryption_algorithm = "AWS-KMS"
+          encryption_key = "Test"
+          access_key = "test"
+          secret_key = "test"
+          with_encryption_key = true;
+          featurestore_id = get_featurestore_id(project.id)
+          json_result, connector_name = create_s3_connector_with_encryption(project.id, featurestore_id, with_encryption_key,
+                                                            encryption_algorithm, encryption_key,access_key,
+                                                                            secret_key, bucket: "testbucket")
+          parsed_json = JSON.parse(json_result)
+          expect_status(201)
+          expect(parsed_json.key?("id")).to be true
+          expect(parsed_json.key?("name")).to be true
+          expect(parsed_json.key?("description")).to be true
+          expect(parsed_json.key?("storageConnectorType")).to be true
+          expect(parsed_json.key?("featurestoreId")).to be true
+          expect(parsed_json.key?("bucket")).to be true
+          expect(parsed_json.key?("secretKey")).to be true
+          expect(parsed_json.key?("accessKey")).to be true
+          expect(parsed_json["name"] == connector_name).to be true
+          expect(parsed_json["storageConnectorType"] == "S3").to be true
+          expect(parsed_json["bucket"] == "testbucket").to be true
+          expect(parsed_json["serverEncryptionAlgorithm"] == encryption_algorithm).to be true
+          expect(parsed_json["serverEncryptionKey"] == encryption_key).to be true
+        end
+
+        it "should be not able to add s3 connector to the featurestore with wrong encryption algorithm" do
+          project = get_project
+          encryption_algorithm = "WRONG-ALGORITHM"
+          encryption_key = "Test"
+          access_key = "test"
+          secret_key = "test"
+          with_encryption_key = true;
+          featurestore_id = get_featurestore_id(project.id)
+          json_result, connector_name = create_s3_connector_with_encryption(project.id, featurestore_id, with_encryption_key,
+                                                                            encryption_algorithm, encryption_key,
+                                                                            access_key, secret_key,
+                                                                            bucket: "testbucket")
+          parsed_json = JSON.parse(json_result)
+          expect_status(400)
+          expect(parsed_json.key?("errorCode")).to be true
+          expect(parsed_json.key?("errorMsg")).to be true
+          expect(parsed_json.key?("usrMsg")).to be true
+          expect(parsed_json["errorCode"] == 270107).to be true
+        end
+
+        it "should be not able to add s3 connector to the featurestore with wrong server key and access key pair" do
```

Comment:
Can you also add a test to check that you can add a storage connector without access/secret key if the iam role variable is set to true.
And also another tests that fails if you add a storage connector without access/secret key and you don't have the IAM role variable set to true.